index.d.ts

```diff
@@ -0,0 +1,104 @@
+// TypeScript Version: 3.7
+
+import {Test} from 'unist-util-is'
+import * as unist from 'unist'
+
+export {}
+
+type Null = undefined | null
+
+declare namespace messageControl {
+  export {}
+  /** a comment marker */
+  export interface Marker<N extends unist.Node> {
+    /** name of marker */
+    name: string
+    /** value after name */
+    attributes: string
+    /** parsed attributes */
+    parameters: Record<string, unknown>
+    /** reference to given node */
+    node: N
+  }
+
+  /** parse a possible comment marker node to a Marker */
+  export interface MarkerFn<N extends unist.Node> {
+    (node: N): Marker<N> | Null
+  }
+
+  interface MessageControlOptionsWithReset<T extends unist.Node>
```

Comment:
I don't think we need separated interfaces based on `reset` option.
 `enable` and `disable` are optional regardless they are used or not.

Reply:
I don't really understand. the reset option is documented to change the interpretation of other fields, which this expresses. Doing it this way means a discriminated union can intuit the state of the other fields. see: https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions

Reply:
The spec documentation doesn't said `disabled` or `enabled` must not be provided if `reset` is `true`, `false` or `undefined`.

Reply:
Catching up on my types here ... I have `enable?: undefined` (should maybe be `undefined | null`), but that just specifies that if enable *is* specified it should be undefined (i.e. not that the field should not be present, only that if it is present it should be `undefined`), which I think makes sense for ensuring well-formed `MessageControlOptions`. This doesn't mean that `MessageControlOptions` can't have `enable` as some other value, but in this case you'd need to do a type guard to extract that.

I'd really much rather communicate this relationship through types than rely on documentation, so maybe setting these to `unknown` when they have no defined meaning in the spec is a good halfway house?